src/codegen.cpp

```diff
@@ -4484,6 +4523,28 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
         return true;
     }
 
+    else if (f == jl_builtin_memorynew && (nargs == 2)) {
+        const jl_cgval_t &memty = argv[1];
+        if (!memty.constant)
+            return false;
+        jl_datatype_t *typ = (jl_datatype_t*) memty.constant;
+        if (!jl_is_concrete_type((jl_value_t*)typ) || !jl_is_genericmemory_type(typ))
+            return false;
+        jl_genericmemory_t *inst = (jl_genericmemory_t*)((jl_datatype_t*)typ)->instance;
+        if (inst == NULL)
+            return false;
+        if(argv[2].constant) {
+            if (argv[2].typ != (jl_value_t*)jl_long_type)
+                return false;
+            size_t nel = jl_unbox_long(argv[2].constant);
+            if (nel < 0)
+                return false;
+            *ret = emit_const_len_memorynew(ctx, typ, nel, inst);
+        } else
+            *ret = emit_memorynew(ctx, typ, argv[2], inst);
```

Comment:
style nit: usually we brace either all or none, like this
```suggestion
        }
        else {
            *ret = emit_memorynew(ctx, typ, argv[2], inst);
        }
```